### PR TITLE
AWS focussed CLI implementation for Shareable Viz 

### LINKED ISSUE #1636: AWS focused Shareable Viz CLI
## Description

We want to add a CLI command to allow users to access the Host Kedro Viz functionality via the Cli. Currently, this feature only supports end to end deployment using AWS, so the CLI is also AWS specific. 

### Prototype
```
kedro viz deploy 

AWS Bucket Info
===============
Bucket Region: 
Bucket Name: 

...creating your webpage (23s) 

Success! Kedro Viz has been deployed. It can be accessed at: 
https://kedroviz-repo/project-123/folder-1/my-hosted-viz-pipeline-quantumblack
```

This interactive flow would be supported by two flags 
```
--region            allow users to specify region inline
--bucket-name       allow users to specify bucket name inline 
```

Example
``` 
kedro viz deploy --region=eu_west_2 --bucket-name=hosted-viz

...creating your webpage (23s) 

Success! Kedro Viz has been deployed. It can be accessed at: 
https://kedroviz-repo/project-123/folder-1/my-hosted-viz-pipeline-quantumblack

```

### If the user hasn't set up credentials 

```
kedro viz deploy 

Deploying and hosting Kedro-Viz requires AWS access keys. 
To use this feature, please supply your AWS access keys as environment variables. 
More information can be found at: https://kedro-docs.org/visualisation/share_kedro_viz.html

```

## Possible Implementation
- this feature is currently blocked by #1616 as `deploy` would need to be added to the `kedro viz` command group
- it would be nice if there was a incrementing time ticker while the site deploys so that the user knows the process isn't hung. `...creating your webpage (23s) `
- alternatively a loading spinner, but i don't know if our CLI supports that. 

## Acceptance Criteria 
- [ ] working `kedro viz deploy` command that triggers interactive flow asking for AWS info  
- [ ] successfully deploys kedro viz to specified s3 bucket 
- [ ] returns active link to hosted site 
- [ ] working `--region` and `--bucket-name` flags, with basic input validation, no verification
- [ ] update the `kedro viz` help text to include the `deploy` command 
`deploy      Deploy Kedro Viz to a hosted link with AWS `

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #1636 

## Development notes

* Add `kedro viz deploy` as a deploy command for shareable viz focussing on AWS
 
shareableviz_deploy_cli

## QA notes

* Install editable kedro viz via `pip install -e package`
* `cd demo-project`
* Run `kedro viz deploy`

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

File: package/kedro_viz/launchers/cli.py [hunk 1]
```diff
@@ -187,3 +195,68 @@ def viz(
...
+                "/UserGuide/Concepts.RegionsAndAvailabilityZones.html"
+                "#Concepts.RegionsAndAvailabilityZones.Regions",
+                fg="red",
+            ),
+        )
+        return
+
+    try:
+        # Loads and populates data from underlying Kedro Project
+        load_and_populate_data(Path.cwd(), ignore_plugins=True)
```
[the remark this answers is not in the document]
Comment (ravi-kumar-pilla) [reply package/kedro_viz/launchers/cli.py:230]: This is to catch any exception in load_and_populate_data

**Review by stephkaiser — APPROVED** [review]

**Review by NeroOkwa — APPROVED** [review]
LGTM!!!

**Review by rashidakanchwala — APPROVED** [review]
Amazing! Thanks @ravi-kumar-pilla

File: docs/source/share_kedro_viz.md [hunk 2]
```diff
@@ -92,6 +92,12 @@ Here's an example of the flow:
 
 ![](./images/kedro-publish-share.gif)
 
+From Kedro-Viz version 7.0.0, you can now publish and share your Kedro-Viz project from the command line. Use the following command from the root folder of your Kedro project
```
Comment (merelcht) [inline docs/source/share_kedro_viz.md:95]:
```suggestion
From Kedro-Viz version 7.0.0, you can publish and share your Kedro-Viz project from the command line. Use the following command from the root folder of your Kedro project
```

File: package/kedro_viz/integrations/deployment/s3_deployer.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def _ingest_heap_analytics(self):
         env = Environment(loader=FileSystemLoader(_HTML_DIR))
 
         if should_add_telemetry:
-            logger.debug("Ingesting heap analytics.")
+            logger.info("Ingesting heap analytics.")
```
Comment (merelcht) [inline package/kedro_viz/integrations/deployment/s3_deployer.py:66]: Should this be a info message?

**Review by merelcht — APPROVED** [review]
LGTM 👍 I was wondering though why several of the logging messages are now `info` instead of `debug`?

**Review by jitu5 — APPROVED** [review]

**Pull request merged**